modules/launcher/src/main/java/org/mule/module/launcher/log4j2/LoggerContextCache.java

```diff
@@ -255,12 +257,16 @@ public void run()
         }
     }
 
-    private int computeKey(ClassLoader classLoader)
+    private String computeKey(ClassLoader classLoader)
     {
-        return classLoader.hashCode();
+        if (classLoader instanceof ApplicationClassLoader)
```

Comment:
What happens when an application is redeployed? and what happens with domains or any other type or artifacT?